src/components/header.js

```diff
@@ -39,40 +39,72 @@ function Header({ slug }) {
   const [isOpen2, setIsOpen2] = useState(false);
 
   /* first dropdown items */
-  const navDropdownLinks1 = ['link-one', 'link-two']
+  const navDropdownLinks1 = ["link-one", "link-two"];
   const navDropdownLinkLabels1 = t("header.nav.dropdowns")[0]["simpleLinks"];
-  const navDropdownLength1 = navDropdownLinks1.length;
-  const navDropdownItems1 = Array.from({ length: navDropdownLength1 }, (_, i) => {
-    return (
-      <Link to={navDropdownLinks1[i]} key={i}>
-        {navDropdownLinkLabels1[i]}
-      </Link>
-    );
-  });
+  const navDropdownLength1 =
+    navDropdownLinks1.length > navDropdownLinkLabels1.length
+      ? navDropdownLinkLabels1.length
+      : navDropdownLinks1.length;
+  const navDropdownItems1 = Array.from(
```

Comment:
Can you map through navDropdownLinks instead of using this syntax? If you use map then you also won't need to compare the array lengths above

Reply:
I compare the array lengths to take the shorter of the two in case the user forgets to add a label in the translation file or add a link to the array. Mapping through the navDropdownLinks can also be done, but this syntax is easier to understand if taking the shorter of the two. If you look at the `parentLinks` mapping (which you want me to do), it's more confusing to understand. Let me know which one you think I should stick to – `Array.from` or `.map`.

Reply:
I personally think map is easier to understand, especially since you won't have to reference the index. 

If the user does forget to add a label or file, then the labels and files might be mismatched, if we proceed with looping through the shorter array. Maybe we could show an error if the lengths don't match?

It is definitely great that you're thinking about cases of user error though!

Reply:
How would we show an error if the lengths don't match? I think what we're doing right now is good enough. I'll let you know once the mapping is complete.

Reply:
If they missed the first link, then all of the link / name pairs would be off. This is actually a good reason to have the links and the labels stored in the same place in an array of objects maybe?

Reply:
If we wanted to show an error we could do something like [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw). But you're right that it's probably not needed now

Reply:
Mapping is complete.

Reply:
> If they missed the first link, then all of the link / name pairs would be off. This is actually a good reason to have the links and the labels stored in the same place in an array of objects maybe?

Are you thinking about doing this in the translation files? If so, what if the user makes a mistake when typing in the link in a random translation file, while the others are correct? This won't stand out to the user and may be dismissed, while the current approach makes sure the links are intact. If they cannot see one of their links in a different language then they will know they messed up with the labeling and their mistake will be apparent, but making sure the links are correct across translation files is a hassle and won't be apparent to the user unless they dig deep.

If you're thinking about doing this in the `link.js` constant file, then this might throw the user off a bit since they will now have to translate in 3 different places (translation files, markdown files, and constant file).

| Method | Pros | Cons |
| Labels in Translation File + Links in Constant File | No data duplication, possible errors are more apparent to the user  | Some room for user error |
| Labels + Links in Translation File | | Data duplication, links can be off and won't be apparent to the user, more room for user error |
| Labels + Links in Constant File | No data duplication, possible errors are more apparent to the user | Some room for user error, translating in 3 different places (i18n translation, markdown, constant files) |

Reply:
What if you use i18next in the `link.js` constant file for the labels

Reply:
So you mean using `t()` in `link.js` to get the translations? It's a constant, and i18next will change the labels on every language change. Should we still do that?

Reply:
Oh do you mean is it bad practice to use a variable in a constants file? I guess the link is the actual constant that we're worried about, but we still want the label to change for each language? If you want to address the potential mismatch I think this makes sense. 

Could we pull the links into the translation.json file from constants/links? I'm not sure if you can import things into a json file

Reply:
Whatever is in a constant file isn't mean to change at all, and by using i18next in the constant file, we're going against this. Not only this but will what you're thinking of work? Also, your idea is trying to help the user make sure they don't make any mistakes in the labeling, but they're going to be working in the translation files anyways, so is this really that necessary?

I don't think you can import things into a JSON file; Google is super unhelpful, at least.

Reply:
Okay. Mostly, I am still just unsure of going ahead and pushing the labels/links in even if the array lengths don't match

Reply:
Console warning messages added!